server/scripts/2020TaxSeasonDataScript.php

```diff
@@ -0,0 +1,409 @@
+	die('SUCCESS');
+}
+
+// Need to create the new sites
+// Create Shifts for each site
+// Create appointment times within those shifts
+// Create role limits for that site/shift/role
+
+function insertAndersonLibraryData() {
+	GLOBAL $DB_CONN;
+
+	$siteCoordinatorRoleId = 1;
+	$greeterRoleId = 2;
+	$preparerRoleId = 3;
+    $reviewerRoleId = 4;
+    
+	$dataAlreadyInserted = true;
+	if ($dataAlreadyInserted) {
+		die('The AL data has already been inserted');
+	}
+
+	$siteId = 2; // Manually obtained from PROD DB
+
+	try {
+		$DB_CONN->beginTransaction();
+
+		// Wednesdays
+		$dates = getWeeklyDatesFromRange('2020-01-29', '2020-04-08');
+		foreach ($dates as $date) {
+			$shiftId = insertShift("$date 17:00:00", "$date 20:00:00", $siteId);
+
+            // Appointment numbers are from per Katie's parameter sheet
+			$firstAppointmentTimeId = insertAppointmentTime("$date 17:00:00", 9, 9, 100, 60, $siteId);
+            $secondAppointmentTimeId = insertAppointmentTime("$date 18:00:00", 9, 9, 100, 60, $siteId);
+            $thirdAppointmentTimeId = insertAppointmentTime("$date 19:00:00", 5, 5, 100, 60, $siteId);
+        }
+        
+        // Thursdays
+        $thursdayDates = getWeeklyDatesFromRange('2020-01-30', '2020-04-09');
+        foreach ($dates as $date) {
+			$shiftId = insertShift("$date 17:00:00", "$date 20:00:00", $siteId);
+
+			$firstAppointmentTimeId = insertAppointmentTime("$date 17:00:00", 6, 6, 100, 60, $siteId);
+            $secondAppointmentTimeId = insertAppointmentTime("$date 18:00:00", 6, 6, 100, 60, $siteId);
+            $thirdAppointmentTimeId = insertAppointmentTime("$date 19:00:00", 5, 5, 100, 60, $siteId);
+        }
+
+		// Default Site Role Limits
+		insertSiteRoleLimit(1, $siteCoordinatorRoleId, $siteId);
+		insertSiteRoleLimit(1, $greeterRoleId, $siteId);
+
+		$DB_CONN->commit();
+	} catch (Exception $e) {
+		$DB_CONN->rollback();
+		throw new Exception('Failed inserting Anderson Library data', MY_EXCEPTION);
+		die();
+	}
+}
+
+function insertCenterForPeopleInNeedData() {
+	GLOBAL $DB_CONN;
+
+	$siteCoordinatorRoleId = 1;
+	$greeterRoleId = 2;
+	$preparerRoleId = 3;
+    $reviewerRoleId = 4;
+	
+	$dataAlreadyInserted = true;
+	if ($dataAlreadyInserted) {
+		die('The Center for People in Need data has already been inserted');
+	}
+
+	try {
+		$DB_CONN->beginTransaction();
+
+        $siteId = 5; // Manually obtained from PROD DB
+
+		// Tuesdays
+		$dates = getWeeklyDatesFromRange('2020-02-04', '2020-03-24');
+		foreach ($dates as $date) {
+			$shiftId = insertShift("$date 11:00:00", "$date 14:00:00", $siteId);
+
+			$firstAppointmentTimeId = insertAppointmentTime("$date 11:00:00", 8, 8, 100, 60, $siteId);
+			$secondAppointmentTimeId = insertAppointmentTime("$date 12:00:00", 8, 8, 100, 60, $siteId);
+			$thirdAppointmentTimeId = insertAppointmentTime("$date 13:00:00", 4, 4, 100, 60, $siteId);
+		}
+
+		// Default Site Role Limits
+		insertSiteRoleLimit(1, $siteCoordinatorRoleId, $siteId);

```

Comment:
On a similar note to the one above, we should verify with Katie that she doesn't want a preparer limit on these sites. Currently, even if there was 50 prepares, we would only let 9 appointments during a time slot, which seems maybe wrong?

Reply:
The preparer limit might very well be the number of prepares she's specified. But also, there's a chance they're moving away from using the "sign up for a shift" approach, so we should verify this all with Katie

Reply:
Did not add anything. "Of that same vein, if we are having volunteers sign up, did you want us to have a preparer limit on the site that is just the number of preparers you specified on the excel sheet? N/A"